modules/RTC/TraceablePeerConnection.js

```diff
@@ -1477,30 +1477,31 @@ TraceablePeerConnection.prototype.addTrack = function(track, isInitiator = false
  * Adds local track as part of the unmute operation.
  * @param {JitsiLocalTrack} track the track to be added as part of the unmute
  * operation
- * @return {boolean} <tt>true</tt> if the state of underlying PC has changed and
- * the renegotiation is required or <tt>false</tt> otherwise.
+ * @return {Promise<boolean>} Promise that resolves to true if the underlying PeerConnection's
+ * state has changed and renegotiation is required, false if no renegotiation is needed or
+ * Promise is rejected when something goes wrong.
  */
 TraceablePeerConnection.prototype.addTrackUnmute = function(track) {
+    if (browser.usesUnifiedPlan()) {
+        return this.tpcUtils.addTrackUnmute(track);
+    }
     if (!this._assertTrackBelongs('addTrackUnmute', track)) {
         // Abort
-        return false;
+        return Promise.reject();
     }
 
     logger.info(`Adding ${track} as unmute to ${this}`);
-    if (browser.usesUnifiedPlan()) {
-        return this.tpcUtils.addTrackUnmute(track);
-    }
     const webRtcStream = track.getOriginalStream();
 
     if (!webRtcStream) {
         logger.error(
             `Unable to add ${track} as unmute to ${this} - no WebRTC stream`);
 
-        return false;
+        return Promise.reject();
```

Comment:
can you also pass an error to this and other rejects?

Reply:
either that or implement a reject handler here: https://github.com/jitsi/lib-jitsi-meet/pull/1078/files#diff-f439e8319bfd0bb8b632a895127bde3eR2018 so it would not count as success